imports/wasi_snapshot_preview1/poll.go

```diff
@@ -142,9 +142,10 @@ func processFDEvent(mod api.Module, eventType byte, inBuf []byte) syscall.Errno
 	// files.
 	errno := syscall.ENOTSUP
 	if eventType == wasip1.EventTypeFdRead {
-		if _, ok := fsc.LookupFile(fd); ok {
-			// If fd is a pipe, IsTerminal is false.
-			if platform.IsTerminal(uintptr(fd)) {
+		if f, ok := fsc.LookupFile(fd); ok {
+			st, _ := f.Stat()
+			// if fd is a pipe, then it is not a char device (a tty)
+			if st.Mode&fs.ModeCharDevice != 0 {
```

Comment:
here I am also proposing to Stat() instead of calling IsTerminal() , but I am not sure which one is cheaper

Reply:
The mode of a file would never change so you could potentially cache it to amortize then stat call. 

I'd leave this to a later update when you have data to demonstrate that the optimization is needed.

Reply:
after verifying the original issue that led to platform.IsTerminal works this way, go back and delete platform.IsTerminal! https://github.com/tetratelabs/wazero/pull/935 was an issue raised both in python and RethinkRaw, so maybe manually check both (especially as there's now a working python build!)

Reply:
indeed, this works with Python (which I checked manually, will check RethinkRaw too)

Reply:
checked both and they seem to work, but I'll ping @ncruces to make sure :) would you like to check? 🙏

Reply:
Oh, hey… users! I'll check for sure, thanks for the heads up.

I really need to spin off `dcraw` into a package with unit tests.

Reply:
@evacchi all good, it seems.